### PR TITLE
feat: Add ButtonIcon Tooltip

### LINKED ISSUE #423: Button Icon optional Tooltip
I think it might be a good idea to show an optional tooltip for Button Icons, so we can provide a quick description for the button's action.

### PR DESCRIPTION
**Pull request by victorgcramos**
Closes #423 

This diff adds an optional tooltip for ButtonIcon.

#### Usage

```javascript
<ButtonIcon
  type="create"
  tooltipText="Create a New Wallet"
  tooltipPlacement="bottom"
/>
```

#### Preview
Screen Shot 2022-04-07 at 12 45 24 PM

### CODE REVIEW

File: src/components/ButtonIcon/ButtonIcon.jsx [hunk 1]
```diff
@@ -3,13 +3,23 @@ import React from "react";
...
   DEFAULT_DARK_THEME_NAME,
 } from "../../theme";
 
+const Wrapper = ({ tooltipText, tooltipPlacement, children }) =>
+  tooltipText ? (
+    <Tooltip content={tooltipText} placement={tooltipPlacement}>
+      {children}
+    </Tooltip>
+  ) : (
+    <>{children}</>
```
Comment (tiagoalvesdulce) [inline src/components/ButtonIcon/ButtonIcon.jsx:20]: no need for the React fragment (`<></>`) here

File: src/components/ButtonIcon/ButtonIcon.jsx [hunk 2]
```diff
@@ -74,6 +90,7 @@ ButtonIcon.propTypes = {
   onClick: PropTypes.func,
   loading: PropTypes.bool,
   text: PropTypes.string,
+  tooltipText: PropTypes.string,
```
Comment (tiagoalvesdulce) [inline src/components/ButtonIcon/ButtonIcon.jsx:93]: I think tooltipPlacement should be added to the propTypes too, right?

File: src/components/Tooltip/Tooltip.jsx [hunk 3]
```diff
@@ -44,7 +45,7 @@ const Tooltip = ({
 Tooltip.propTypes = {
   children: PropTypes.node,
   content: PropTypes.node,
-  placement: PropTypes.string,
+  placement: PropTypes.oneOf(["top", "bottom", "right", "left"]),
```
Comment (tiagoalvesdulce) [inline src/components/Tooltip/Tooltip.jsx:48]: 👍

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM. left a few inline comments.

File: src/components/ButtonIcon/ButtonIcon.jsx (hunk 2 above)
Comment (victorgcramos) [reply src/components/ButtonIcon/ButtonIcon.jsx:93]: Sure!

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM

**Pull request merged**